src/pybind/mgr/dashboard/run-frontend-e2e-tests.sh

```diff
@@ -10,78 +29,92 @@ stop() {
     exit $1
 }
 
-BASE_URL=''
+check_devive_available() {
+    : ${DEVICE:="chrome"}
+    failed=false
+
+    if [ "$DEVICE" == "docker" ]; then
+        [ -x "$(command -v docker)" ] || failed=true
+    else
+        cd $DASH_DIR/frontend
+        npx cypress verify
+
+        case "$DEVICE" in
+            chrome)
+                [ -x "$(command -v google-chrome)" ] || [ -x "$(command -v google-chrome-stable)" ] ] || failed=true
+                ;;
+            chromium)
+                [ -x "$(command -v chromium)" ] || failed=true
+                ;;
+        esac
+    fi
+
+    if [ "$failed" = "true" ]; then
+            echo "ERROR: $DEVICE not found. You need to install $DEVICE or \
+    use a different device. Supported devices: electron (default), chrome, chromium or docker."
+        stop 1
+    fi
+}
+
+CYPRESS_BASE_URL=''
 DEVICE=''
-E2E_LOGIN_USER=''
-E2E_LOGIN_PWD=''
+CYPRESS_LOGIN_PWD=''
+CYPRESS_LOGIN_USER=''
+NO_COLOR=1
 REMOTE='false'
 
 while getopts 'd:p:r:u:' flag; do
```

Comment:
It would be great to have the possibility of passing ```--spec <file>``` to use along with ```it.only```
when you're interested in running only 1 test:
https://docs.cypress.io/guides/guides/command-line.html#cypress-run-spec-lt-spec-gt
https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Excluding-and-Including-Tests

Reply:
I think the unittest and e2e scripts should be focused on running the complete test.
If you want to run just 1 spec, using the cypress app is the best option.